src/tests/suite/dfuse_test.c

```diff
@@ -315,6 +315,120 @@ timespec_gt(struct timespec t1, struct timespec t2)
 }
 
+void
+do_mtime(void **state)
+{
+	struct stat     stbuf;
+	struct timespec prev_ts;
+	struct timespec now;
+	struct timespec times[2];
+	int             fd;
+	int             rc;
+	char            input_buf[] = "hello";
+	int             root        = open(test_dir, O_PATH | O_DIRECTORY);
+
+	assert_return_code(root, errno);
+
+	/* Open a file and sanity check the mtime */
+	fd = openat(root, "my_file", O_RDWR | O_CREAT | O_EXCL, S_IWUSR | S_IRUSR);
+	assert_return_code(fd, errno);
+	rc = clock_gettime(CLOCK_REALTIME, &now);
+	assert_return_code(fd, errno);
+	rc = fstat(fd, &stbuf);
+	assert_return_code(rc, errno);
+	prev_ts.tv_sec  = stbuf.st_mtim.tv_sec;
+	prev_ts.tv_nsec = stbuf.st_mtim.tv_nsec;
+	assert_true(now.tv_sec - prev_ts.tv_sec < 3);
+
+	/* Write to the file and verify mtime is newer */
+	rc = write(fd, input_buf, sizeof(input_buf));
+	assert_return_code(rc, errno);
+	rc = fstat(fd, &stbuf);
+	assert_return_code(rc, errno);
+	assert_true(timespec_gt(stbuf.st_mtim, prev_ts));
+	prev_ts.tv_sec  = stbuf.st_mtim.tv_sec;
+	prev_ts.tv_nsec = stbuf.st_mtim.tv_nsec;
+
+	/* Truncate the file and verify mtime is newer */
+	rc = ftruncate(fd, 0);
+	assert_return_code(rc, errno);
+	rc = fstat(fd, &stbuf);
+	assert_return_code(rc, errno);
+	assert_true(timespec_gt(stbuf.st_mtim, prev_ts));
+	prev_ts.tv_sec  = stbuf.st_mtim.tv_sec;
+	prev_ts.tv_nsec = stbuf.st_mtim.tv_nsec;
+
+	/* Set and verify mtime set in the past */
+	times[0]         = now;
+	times[1].tv_sec  = now.tv_sec - 10;
+	times[1].tv_nsec = 20;
+	rc               = futimens(fd, times);
+	assert_return_code(fd, errno);
+	rc = fstat(fd, &stbuf);
+	assert_return_code(rc, errno);
+	assert_int_equal(stbuf.st_mtim.tv_sec, times[1].tv_sec);
+	assert_int_equal(stbuf.st_mtim.tv_nsec, times[1].tv_nsec);
+	prev_ts.tv_sec  = stbuf.st_mtim.tv_sec;
+	prev_ts.tv_nsec = stbuf.st_mtim.tv_nsec;
+
+	rc = close(fd);
+	assert_return_code(rc, errno);
+
+	rc = unlinkat(root, "my_file", 0);
+	assert_return_code(rc, errno);
+
+	rc = close(root);
+	assert_return_code(rc, errno);
+}
+
+static int
+run_specified_tests(const char *tests, int *sub_tests, int sub_tests_size)
+{
+	int nr_failed = 0;
+
+	if (strlen(tests) == 0)
+		tests = all_tests;
+
+	while (*tests != '\0') {
+		switch (*tests) {
+		case 'i':
+			printf("\n\n=================");
+			printf("dfuse IO tests");
+			printf("=====================");
+			const struct CMUnitTest io_tests[] = {
+			    cmocka_unit_test(do_openat),
+			    cmocka_unit_test(do_ioctl),
+			};
+			nr_failed = cmocka_run_group_tests(io_tests, NULL, NULL);
+			break;
+		case 'm':
+			printf("\n\n=================");
+			printf("dfuse metadata tests");
+			printf("=====================");
+			const struct CMUnitTest metadata_tests[] = {
+			    cmocka_unit_test(do_mtime),
+			};
+			nr_failed = cmocka_run_group_tests(metadata_tests, NULL, NULL);
+			break;
+
+		default:
+			assert_true(0);
+		}
+
+		tests++;
+	}
+
+	return nr_failed;
+}
+
+=======
```

Comment:
(style)  spaces required around that '==' (ctx:ExO)
(style)  spaces required around that '==' (ctx:OxO)
(style)  spaces required around that '==' (ctx:OxO)
(style)  spaces required around that '=' (ctx:OxE)